benchmark/cajita/Cajita_InterpolationPerformance.cpp

```diff
@@ -0,0 +1,402 @@
+                p2g_vector_value_timer.start( n );
+                p2g( vector_p2g, position, position.size(), Spline<1>(),
+                     *vector_halo, *vector_grid_field );
+                p2g_vector_value_timer.stop( n );
+
+                // P2G scalar gradient
+                auto scalar_grad_p2g = createScalarGradientP2G( scalar, -0.5 );
+                p2g_scalar_gradient_timer.start( n );
+                p2g( scalar_grad_p2g, position, position.size(), Spline<1>(),
+                     *vector_halo, *vector_grid_field );
+                p2g_scalar_gradient_timer.stop( n );
+
+                // P2G vector divergence
+                auto vector_div_p2g = createVectorDivergenceP2G( vector, -0.5 );
+                p2g_vector_divergence_timer.start( n );
+                p2g( vector_div_p2g, position, position.size(), Spline<1>(),
+                     *scalar_halo, *scalar_grid_field );
+                p2g_vector_divergence_timer.stop( n );
+
+                // P2G tensor divergence
+                auto tensor_div_p2g = createTensorDivergenceP2G( tensor, -0.5 );
+                p2g_tensor_divergence_timer.start( n );
+                p2g( tensor_div_p2g, position, position.size(), Spline<1>(),
+                     *vector_halo, *vector_grid_field );
+                p2g_tensor_divergence_timer.stop( n );
+
+                // G2P scalar value
+                auto scalar_value_g2p = createScalarValueG2P( scalar, -0.5 );
+                g2p_scalar_value_timer.start( n );
+                g2p( *scalar_grid_field, *scalar_halo, position,
+                     position.size(), Spline<1>(), scalar_value_g2p );
+                g2p_scalar_value_timer.stop( n );
+
+                // G2P vector value
+                auto vector_value_g2p = createVectorValueG2P( vector, -0.5 );
+                g2p_vector_value_timer.start( n );
+                g2p( *vector_grid_field, *vector_halo, position,
+                     position.size(), Spline<1>(), vector_value_g2p );
+                g2p_vector_value_timer.stop( n );
+
+                // G2P scalar gradient
+                auto scalar_gradient_g2p =
+                    createScalarGradientG2P( vector, -0.5 );
+                g2p_scalar_gradient_timer.start( n );
+                g2p( *scalar_grid_field, *scalar_halo, position,
+                     position.size(), Spline<1>(), scalar_gradient_g2p );
+                g2p_scalar_gradient_timer.stop( n );
+
+                // G2P vector gradient
+                auto vector_gradient_g2p =
+                    createVectorGradientG2P( tensor, -0.5 );
+                g2p_vector_gradient_timer.start( n );
+                g2p( *vector_grid_field, *vector_halo, position,
+                     position.size(), Spline<1>(), vector_gradient_g2p );
+                g2p_vector_gradient_timer.stop( n );
+
+                // G2P vector divergence
+                auto vector_div_g2p = createVectorDivergenceG2P( scalar, -0.5 );
```

Comment:
This is nice. Let's add three more where we fuse all G2P operations into a single kernel and single halo, all P2G, and then do a P2G2P or something. I want to be able to measure effects of kernel fusion (e.g. latency improvements for low particle counts) by comparing to the total times of the individual operations.

Reply:
Add those 3 - ready for review, but also still need to post Summit results

Reply:
They look good. Just remember when processing results that the individual kernels you sum for comparison are the same.